packages/dag4-wallet/src/shared/operations.ts

```diff
@@ -0,0 +1,185 @@
+import { keyStore, KeyTrio } from "@stardust-collective/dag4-keystore";
+import {
+  AllowSpendWithCurrencyId,
+  DagNetwork,
+  GlobalDagNetwork,
+  HashResponse,
+  MetagraphTokenNetwork,
+  SignedAllowSpend,
+  SignedTokenLock,
+  TokenLockWithCurrencyId,
+  TokenLockWithParent,
+  TransactionReference,
+} from "@stardust-collective/dag4-network";
+import { normalizePublicKey } from "../utils";
+import {
+  dagAddressValidator,
+  allowSpendSchema,
+  tokenLockSchema,
+  validateArraySchema,
+  validateSchema,
+} from "../validationSchemas";
+import { DagAccount } from "../dag-account";
+
+type SharedNetwork = DagNetwork | GlobalDagNetwork | MetagraphTokenNetwork;
+type OperationType = "allowSpend" | "tokenLock";
+type BodyType = AllowSpendWithCurrencyId | TokenLockWithCurrencyId;
+
+export const executeOperation = async (
```

Comment:
This is a pretty ugly pattern imo where we're passing a string to call a nested function instead of just calling the original function itself. Introducing a dispatcher pattern for just these two methods makes the code less readable and doesn't add much benefit. 

Why not just create a `accountIsActive` helper and call it from both `allowSpend` and `tokenLock` functions?